### ISSUE TITLE
Post editor: The 'Save' keyboard shortcut doesn't work within Patterns

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#51907]
### Description

The Save keyboard shortcut Cmd/Ctrl + S is likely the most known and used one in WordPress and in the editor. As a user, I'd expect to always work or, at least, to always _do something_.

Instead, it appears that when editing a Pattern in the Post Editor, the shortcut doesn't do anything. As a useer, I'd be pretty confused: _suddenly_ with no apparent reason the shortcut no longer works. Did I do something wrong? Did WordPress crashed? What's happening?

I do realize that editing a Pattern triggers the 'entity saving' flow, which is different from the normal saving flow. However, the Save shortcuts that suddenly doesn't do anything feels just broken. I could think ot two options:
- At the very least, the shortcut should open the Save panel and show the entity saving flow.
- Another option, maybe not entirely safe, is to make the shortcut save everything anyways. Assuming pro users would use the save shortcut as a mean to quickly save everything and get rid of any intermediary step.

### Step-by-step reproduction instructions

- Edit a post, add a standard paragraph and a Pattern made, for example, of a group that contains two paragraphs.
- Save as a Draft, 
- Refresh the page.
- Edit something within the Pattern.
- Observe the 'Update...' button shows a dot on the left.
- Press Cmd/Ctrl + S
- Observe that nothing happens: the changes are not saved, there's no feedback at all about what's going on.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
